src/index/base.h

```diff
@@ -117,7 +119,7 @@ class BaseIndex : public CValidationInterface
 
     /// Start initializes the sync state and registers the instance as a
     /// ValidationInterface so that it stays in sync with blockchain updates.
-    void Start();
+    void Start(CChainState& active_chainstate);
```

Comment:
Any reason that you decided to put this in the `Start()` method rather than the constructor. You have access to the `::ChainstateActive()` object at the point you construct the index objects.

Adding this parameter to the ctor would allow you to set `m_chainstate` in the initializer list and make it a const member.

Reply:
I believe I did this mostly because ctors are not inherited, whereas methods are. If I were to make `m_chainstate` be a const member and initialized in the ctor, I'd have to change the ctor of every subclass.